src/test/script_tests.cpp

```diff
@@ -2471,5 +2471,27 @@ BOOST_AUTO_TEST_CASE(bitcoinconsensus_verify_script_invalid_flags)
     BOOST_CHECK_EQUAL(err, bitcoinconsensus_ERR_INVALID_FLAGS);
 }
 
+BOOST_AUTO_TEST_CASE(coloredScripts)
+{
+    std::vector<unsigned char> data;
+    std::vector<unsigned char> colorId;
+
+    // CP2PKH(Colored P2PKH)：
+    // <COLOR identifier> OP_COLOR OP_DUP OP_HASH160 <H(pubkey)> OP_EQUALVERIFY OP_CHECKSIG
+    CScript CP2PKHScriptPubKey = CScript() << ParseHex("011863143c14c5166804bd19203356da136c985678cd4d27a1b8c6329604903262") << OP_COLOR << OP_DUP << OP_HASH160 << ParseHex("1018853670f9f3b0582c5b9ee8ce93764ac32b93") << OP_EQUALVERIFY << OP_CHECKSIG;
+
+    //check CP2PKHScriptPubKey match ColoredPayToPubkeyHash not match ColoredPayToScriptHash
+    // BOOST_CHECK(MatchColoredPayToPubkeyHash(CP2PKHScriptPubKey, data, colorId));
+    BOOST_CHECK(!CP2PKHScriptPubKey.IsColoredPayToScriptHash());
+
+    // CP2SH(Colored P2SH)：
+    // <COLOR identifier> OP_COLOR OP_HASH160 <H(redeem script)> OP_EQUAL
+    CScript CP2SHScriptPubKey = CScript() << ParseHex("011863143c14c5166804bd19203356da136c985678cd4d27a1b8c6329604903262") << OP_COLOR << OP_HASH160 << ParseHex("da8a647bba351bbae4cee0089d373c97ec240580") << OP_EQUAL;
+
+    //check CP2SHScriptPubKey match ColoredPayToScriptHash not match ColoredPayToPubkeyHash
+    // BOOST_CHECK(!MatchColoredPayToPubkeyHash(CP2SHScriptPubKey, data, colorId));
+    BOOST_CHECK(CP2SHScriptPubKey.IsColoredPayToScriptHash());
+}
+
```

Comment:
In order to check the conditions comprehensively, you need to test the following cases

* Check if all token types are valid
* Check when an undefined token type is specified.
* Check for cases with different data lengths.

Reply:
ok